homeassistant/components/gpsd/sensor.py

```diff
@@ -126,11 +154,10 @@ def __init__(
     @property
     def native_value(self) -> str | None:
         """Return the state of GPSD."""
-        if self.agps_thread.data_stream.mode == 3:
-            return "3d_fix"
-        if self.agps_thread.data_stream.mode == 2:
-            return "2d_fix"
-        return None
+        value = self.entity_description.value_fn(self.agps_thread)
+        if value == "n/a":
+            return None
+        return value
```

Comment:
```suggestion
        return self.entity_description.value_fn(self.agps_thread)
```

Reply:
In the beginning, until data appears, the state is "n/a" and I wanted to change it to `None`. Anyways, changed it.